src/Tests/Tests/Test_Vibrato.cpp

```diff
@@ -16,6 +15,314 @@ namespace vibrato_test {
+        float fFrequencyInHertz;
+        float fWidth;
+
+    };
+
+    TEST_F(LFOTest, InitTest)
+    {
+        EXPECT_TRUE(lfoSine);
+        EXPECT_TRUE(lfoDC);
+    }
+
+    TEST_F(LFOTest, WidthTest)
+    {
+        float tolerance = 0.05; 
+        float maxValue = 0;
+        float auxValue = 0;
+        int lfoArrayLength = lfoSine->getLength();
+        float* pWavetableArray_1 = new float[lfoArrayLength];
+        
+        for (int i = 0; i < lfoArrayLength; i++) {
+            auxValue = lfoSine->readSample();
+            if (auxValue > maxValue)
+                maxValue = auxValue;
+        }
+        EXPECT_NEAR(maxValue, fWidth, tolerance);
+
+        float testFreq = 5;
+        lfoSine->setFreq(testFreq);
+        int lfoArrayLengthTest2 = static_cast<int> (lfoArrayLength / (fFrequencyInHertz / testFreq));
+        float* pWavetableArray_2 = new float[lfoArrayLengthTest2];
+
+        auxValue = 0;
+        maxValue = 0;
+        for (int i = 0; i < lfoArrayLengthTest2; i++) {
+            auxValue = lfoSine->readSample();
+            if (auxValue > maxValue)
+                maxValue = auxValue;
+        }
+        EXPECT_NEAR(maxValue, fWidth, tolerance);
+        
+
+        delete[] pWavetableArray_1;
+        delete[] pWavetableArray_2;
+    }
+
+
+    class RingBufferTest : public ::testing::Test{
+    protected:
+        // You can remove any or all of the following functions if their bodies would
+        // be empty.
+
+        RingBufferTest() {
+            // You can do set-up work for each test here.
+        }
+
+        ~RingBufferTest() override {
+            // You can do clean-up work that doesn't throw exceptions here.
+        }
+
+        // If the constructor and destructor are not enough for setting up
+        // and cleaning up each test, you can define the following methods:
+
+        void SetUp() override {
+            // Code here will be called immediately after the constructor (right
+            // before each test).
+            ringBuffer = new CRingBuffer<float>(5);
+            for(int i=0;i<ringBuffer->getLength();i++)
+            {
+                ringBuffer->putPostInc(i+1);
+            }
+            ringBuffer->setReadIdx(0);
+        }
+
+        void TearDown() override {
+            // Code here will be called immediately after each test (right
+            // before the destructor).
+            delete ringBuffer;
+        }
+
+        // Class members declared here can be used by all tests in the test suite
+        CRingBuffer<float> *ringBuffer;
+
+    };
+
+    TEST_F(RingBufferTest, InterpolationTest)
+    {
+        EXPECT_NEAR(ringBuffer->get(1.5f), 2.5f,1e-4f);
+    }
+
+    TEST_F(RingBufferTest, WrapAroundTest)
+    {
+        for(int i=0;i<ringBuffer->getLength();i++)
+        {
+            ringBuffer->getPostInc();
+        }
+        EXPECT_EQ(ringBuffer->getReadIdx(), 0);
+        EXPECT_EQ(ringBuffer->getWriteIdx(), 0);
+    }
 }
 
```

Comment:
Also maybe read/write pointer check can be added to ensure no over run/under run issue.